first version of AI annotation for the codebase (#19772)
ydb-platform/ydb: AGENTS.md

> # Project Agents.md Guide for AI agents
> 
> This file provides comprehensive guidance for AI agents working with this codebase.
> 
> ## More information about the project
> 
> The documentation is available here: https://ydb.tech/docs/en/.
> 
> ## Project Structure for AI agents Navigation
> 
> Every directory is a project (a library or a program).
> 
> - `/ydb/core`: core modules
> - `/ydb/library`: ydb libraries
> - `/ydb/docs`: ydb documenation
> - `/library`: common libraries, never change them
> - `/util`: common libraries, never change them

Reviewer: ```suggestion
# Project Agents.md Guide for AI Agents

This file provides comprehensive guidance for AI agents working with this codebase.

## More Information About the Project

The documentation is available at <https://ydb.tech/docs/en/>.

## Project Structure for AI Agents Navigation

Every directory is a project (a library or a program).

- `/ydb/core`: Core modules.
- `/ydb/library`: YDB libraries.
- `/ydb/docs`: YDB documentation.
- `/library`: Common libraries; never change them.
- `/util`: Common libraries; never change them.
```

> ## Building and testing requirements for AI agents
> 
> Building and testing are made using ya utility (ya-tool), which located in the root of the codebase: `/ya`.
> 
> To build a project AI agent should run:
> 
> ```bash
> cd path/to/project
> /ya make
> ```
> 
> To test a project AI agent should run:
> 
> ```bash
> cd path/to/project
> /ya make -A
> ```
> 
> The more information is available here: https://ydb.tech/docs/en/contributor/build-ya.

Reviewer: ````suggestion
## Building and Testing Requirements for AI Agents

Building and testing are performed using the Ya utility (`ya-tool`), which is located in the root of the codebase: `/ya`.

To build a project, an AI agent should run:

```bash
cd path/to/project
/ya make
```

To test a project, an AI agent should run:

```bash
cd path/to/project
/ya make -A
```

More information is available at <https://ydb.tech/docs/en/contributor/build-ya>.
````

## Coding Conventions for AI agents

### C++ Standards for AI agents

> - Use modern C++ but not higher than 20 version.

Reviewer: ```suggestion
- Use modern C++ but not higher than 20 version.

### Documentation Guidelines for AI Agents

- Follow the style guide: <https://ydb.tech/docs/en/contributor/documentation/style-guide>
- Keep the structure consistent: <https://ydb.tech/docs/en/contributor/documentation/structure>
```

> ## Coding Conventions for AI agents
> 
> ### C++ Standards for AI agents
> 
> - Use modern C++ but not higher than 20 version.
> 
> ## Pull Request Guidelines for AI agents
> 
> - The description of pull requests should contain changelog entry (short description of the changes) along with more detailed description for reviewers.
> - Specify only one of the following PR categories:
> * New feature
> * Experimental feature
> * Improvement
> * Performance improvement
> * User Interface
> * Bugfix
> * Backward incompatible change
> * Documentation (changelog entry is not required)
> * Not for changelog (changelog entry is not required)

Reviewer: ```suggestion
## Coding Conventions for AI Agents

### C++ Standards for AI Agents

- Use modern C++ (no later than C++20).

## Pull Request Guidelines for AI Agents

- Pull request descriptions must include a changelog entry (a short summary of the changes), followed by a detailed description for reviewers.
- Specify exactly one of the following PR categories:
  - New Feature
  - Experimental Feature
  - Improvement
  - Performance Improvement
  - User Interface
  - Bugfix
  - Backward-Incompatible Change
  - Documentation (changelog entry is not required)
  - Not for Changelog (changelog entry is not required)
```
